### PR TITLE
Migrate from master to main

### PR DESCRIPTION
**Pull request by wredzio**

### CODE REVIEW

File: React/Extensions/HtmlHelperExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.DependencyInjection;
+using React.Services;
+
+namespace React.Extensions;
```
Comment (KonradGalczynski) [inline React/Extensions/HtmlHelperExtensions.cs]: please consider moving the extension method to the namespace of the extended type as this is good practice from a usage perspective

File: React/Extensions/HtmlHelperExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+public static class HtmlHelperExtensions
+{
+    public static async Task<IHtmlContent> ReactAsync<T>(
```
Comment (KonradGalczynski) [inline React/Extensions/HtmlHelperExtensions.cs]: please align naming of the method with its implementation

File: React/Extensions/ReactExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+using ReactForte.Application.Webpack;
+using Microsoft.AspNetCore.Http;
+
+namespace React.Extensions;
+
+public static class ReactExtensions
+{
+    public static void AddReact(
+        this IServiceCollection services, IConfiguration configuration,
+        string rootPath)
```
Comment (KonradGalczynski) [inline React/Extensions/ReactExtensions.cs]: please align vertical formatting for methods names and arguments (sometimes tailing ')' is in a new line, sometimes after last argument, sometimes there is one arg per line, sometimes more)

File: React/Services/ReactForteService.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+internal class ReactForteService : IReactService
+{
+    private readonly IHtmlService _htmlService;
+
+    public ReactForteService(IHtmlService htmlService)
+    {
+        _htmlService = htmlService;
+    }
+
+    public async Task<IHtmlContent> RenderToStringAsync<T>(IHtmlHelper htmlHelper, string componentName, T props)
```
Comment (KonradGalczynski) [inline React/Services/ReactForteService.cs]: as `htmlHelper` is not used in this concrete please consider injecting it via constructor to `ReactDotNetService` and change method to `Task<IHtmlContent> RenderToStringAsync<T>(string componentName, T props)`

File: ReactForte/Application/HtmlService.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+internal class HtmlService : IHtmlService
+{
+    private readonly IReactService _reactService;
+
+    public HtmlService(IReactService reactService)
+    {
+        _reactService = reactService;
+    }
+
```
Comment (KonradGalczynski) [inline ReactForte/Application/HtmlService.cs:20]: please remove spare line

File: ReactForte/Application/Webpack/WebpackManifest.cs [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    public WebpackManifest(IOptions<WebpackOptions> webpackOptions, string contentRootPath)
+    {
+        _webpackOptions = webpackOptions.Value;
+        IManifestLoader manifestLoader = _webpackOptions.IsDevServer
+            ? new DevServerLoader(ManifestPath, _webpackOptions.OutputPath)
+            : new LocalLoader(contentRootPath, _webpackOptions.OutputPath, ManifestPath);
+        _localInstance = new Lazy<IReadOnlyDictionary<string, string>>(() => manifestLoader.LoadAsync().Result);
+    }
+
+    private static IEnumerable<string> FilterUrls(IEnumerable<string> urls, IEnumerable<string> pattern,
```
Comment (KonradGalczynski) [inline ReactForte/Application/Webpack/WebpackManifest.cs]: Please consider using step-down rule to increase readability -> https://dzone.com/articles/the-stepdown-rule

Comment (wredzio) [reply ReactForte/Application/Webpack/WebpackManifest.cs]: I have moved WebpackManifest to alloy-docker project and changed the implementation to 


`
public record WebpackAssets
{
    private readonly IReadOnlyDictionary<string, string> _assets;

    private static readonly IReadOnlyDictionary<WebpackManifestAssetType, string> AssetTypeExtensions =
        new Dictionary<WebpackManifestAssetType, string>
        {
            { WebpackManifestAssetType.Javascript, ".js" },
            { WebpackManifestAssetType.Style, ".css" }
        };

    public WebpackAssets(IReadOnlyDictionary<string, string> assets)
    {
        _assets = assets;
    }

    public WebpackAssets Get(WebpackManifestAssetType assetType)
    {
        return new WebpackAssets(new Dictionary<string, string>(_assets.Where(pair =>
            pair.Value.EndsWith(AssetTypeExtensions[assetType]))));
    }

    public WebpackAssets WithNames(IEnumerable<string> names)
    {
        return new WebpackAssets(new Dictionary<string, string>(_assets.Where(pair => names.Any(pair.Value.Contains))));
    }

    public WebpackAssets Exclude(IEnumerable<string> names)
    {
        return new WebpackAssets(
            new Dictionary<string, string>(_assets.Where(pair => names.Any(pair.Value.Contains) == false)));
    }

    public IList<string> ToList()
    {
        return new List<string>(_assets.Values);
    }
}
`

**Review by konrad-galczynski — APPROVED** [review]

**Pull request merged**